src/uri.cr

```diff
@@ -225,6 +265,79 @@ class URI
     io
   end
 
+  # Returns whether given byte is reserved character defined in RFC 3986.
+  #
+  # Reserved characters are ':', '/', '?', '#', '[', ']', '@', '!',
+  # '$', '&', "'", '(', ')', '*', '+', ',', ';' and '='.
+  def self.reserved?(byte)
+    char = byte.chr
+    '&' <= char <= ',' ||
+      char == '!' ||
+      char == '#' ||
+      char == '$' ||
+      char == '/' ||
+      char == ':' ||
+      char == ';' ||
+      char == '?' ||
+      char == '@' ||
+      char == '[' ||
+      char == ']' ||
+      char == '='
+  end
```

Comment:
i think this can be shorter 
```crystal
('&' <= char <= ',') || {'!', '#', '$', '/', ':', ';', '?', '@', '[', ']', '='}.includes?(char)
```

and equal by performans